### DISCUSSION TITLE
Deploy question regarding using npm specifiers

### DISCUSSION THREAD

**Discussion by R3uan3** [denoland/deno#16039] [Q&A]
I'm trying to learn how to upload my `MongoDB` project to Deno, under deployment I selected:

![image](https://user-images.githubusercontent.com/103058225/192383223-b12c006b-a473-421f-a3d8-3ba84047ce6b.png)

`index.js`:
```js
"use strict"

const Koa = require("koa")
const Bodyparser = require("koa-bodyparser")
const Logger = require("koa-logger")
const Cors = require("@koa/cors")
const Mongo = require("koa-mongo")
const config = require("./config")
const routes = require("./routes")

const app = new Koa()

app.proxy = true

app.use(
    Mongo({
        url: config.mongodb_uri,
        max: 100,
        min: 1
    })
)

app.use(Logger())
app.use(Cors({ origin: "*", credentials: false }))
app.use(Bodyparser({ jsonLimit: "15mb" }))

routes.forEach(route => app.use(route.routes()))

app.listen(config.port, () => {
    console.log(`🚀 API running on ${config.env} port ${config.port}`)
})
```

It is not compiled and I received this error:
![image](https://user-images.githubusercontent.com/103058225/192382939-ff2e4f75-9587-44ce-b692-246961d731eb.png)

What I'm missing?

**Comment by cjihrig** [comment]
`require()` is not currently available in Deploy, as it is still an unstable feature.

Side note: Issues and feedback regarding Deploy should be opened at https://github.com/denoland/deploy_feedback.

**Reply by R3uan3** [reply]
I see, but will it be available after beta?
What should I use instead of require? import?

**Reply by cjihrig** [reply]
I can't say for sure if it will ever be available in Deploy since it's always possible for unstable features to be removed.

`import` will work, but depending on the code that you're importing, that could fail.

**Reply by R3uan3** [reply]
I've modified all files using `require` to `import` and changed on package json `"type": "module"`

index.js
```js
"use strict"

import Koa from 'koa'
import Bodyparser from 'koa-bodyparser'
import Logger from 'koa-logger'
import Cors from '@koa/cors'
import Mongo from 'koa-mongo'
import config from "./config.js"
import routes from "./routes.js"

const app = new Koa()

app.proxy = true

app.use(
    Mongo({
        url: config.mongodb_uri,
        max: 100,
        min: 1
    })
)

app.use(Logger())
app.use(Cors({ origin: "*", credentials: false }))
app.use(Bodyparser({ jsonLimit: "15mb" }))

routes.forEach(route => app.use(route.routes()))

app.listen(config.port, () => {
    console.log(`🚀 API running on ${config.env} port ${config.port}`)
})
```
routes.js
```js
"use strict"

//const Router = require("koa-router")
//const bind = require("koa-clean")
import Router from 'koa-router'
import bind from 'koa-clean'
```
Now I'm getting this error:
`Error The deployment failed: Relative import path "koa-clean" not prefixed with / or ./ or ../`

**Reply by kitsonk** [reply]
> Issues and feedback regarding Deploy should be opened at https://github.com/denoland/deploy_feedback.

Or join the Discord server for more interactive help and support: https://discord.gg/deno
